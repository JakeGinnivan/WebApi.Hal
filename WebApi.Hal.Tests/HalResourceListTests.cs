using System.Collections.Generic;
using System.IO;
using System.Net.Http;
using ApprovalTests;
using ApprovalTests.Reporters;
using WebApi.Hal.Tests.Representations;
using Xunit;

namespace WebApi.Hal.Tests
{
    public class HalResourceListTests
    {
        readonly RepresentationList<OrganisationRepresentation> representation;

        public HalResourceListTests()
        {
            representation = new OrganisationListRepresentation(
                new List<OrganisationRepresentation>
                       {
                           new OrganisationRepresentation(1, "Org1"),
                           new OrganisationRepresentation(2, "Org2")
                       });
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void organisation_list_get_xml_test()
        {
            // arrange
            var mediaFormatter = new XmlHalMediaTypeFormatter();
            var content = new StringContent(string.Empty);
            var type = representation.GetType();

            // act
            using (var stream = new MemoryStream())
            {
                mediaFormatter.WriteToStream(type, representation, stream, content);
                stream.Seek(0, SeekOrigin.Begin);
                var serialisedResult = new StreamReader(stream).ReadToEnd();

                // assert
                Approvals.Verify(serialisedResult);
            }
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void organisation_list_get_json_test()
        {
            // arrange
            var mediaFormatter = new JsonHalMediaTypeFormatter { Indent = true };
            var content = new StringContent(string.Empty);
            var type = representation.GetType();

            // act
            using (var stream = new MemoryStream())
            {
                mediaFormatter.WriteToStreamAsync(type, representation, stream, content, null).Wait();
                stream.Seek(0, SeekOrigin.Begin);
                var serialisedResult = new StreamReader(stream).ReadToEnd();

                // assert
                Approvals.Verify(serialisedResult);
            }
        }
    }
}