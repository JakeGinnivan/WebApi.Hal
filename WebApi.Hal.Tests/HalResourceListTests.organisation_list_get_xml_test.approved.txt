<?xml version="1.0" encoding="utf-8"?>
<resource rel="organisations" href="/api/organisations">
  <resource rel="organisation" href="/api/organisations/1">
    <Id>1</Id>
    <Name>Org1</Name>
  </resource>
  <resource rel="organisation" href="/api/organisations/2">
    <Id>2</Id>
    <Name>Org2</Name>
  </resource>
</resource>